packages/offix-datastore/src/Model.ts

```diff
@@ -134,9 +134,15 @@ export class Model<T = unknown> {
             });
     }
 
-    private async subscribeForServerEvents(replicator: IReplicator) {
+    // TODO this method should be public
+    // so we can then specify the type of subscription event
+    // the replicator in this case should not be passed as a variable
+    private subscribeForServerEvents(replicator: IReplicator) {
```

Comment:
@wtrocki @Eunovo I want to make this method public and maybe change the name. The idea here is that you could subscribe at the app level in a specific view. The issue though is passing the `replicator` to the model.

Reply:
This issue is that the replication engine is initiated after the model has been created, so the as far as I can tell, the replicator is never going to be passed to the constructor method of the model

Reply:
See what I have pointed in previous PR from @Eunovo 
This needs to be public and also needs to accept filter rather than replicator.

Reply:
@wtrocki yeah I left the replicator in for now, but I definitely plan on removing that and passing in the filter and the subscription type. Just not sure yet how we plan on making the replicator available to the model.

Reply:
@kingsleyzissou If you do need to pass replicator to model, there are two things you could do;
- use a callback as I did with Storage
- refactor query builder to build queries, one model, at a time and add them to the replicator using a public function like `replicator.addQuery(storeName, query)`(which internally builds the query map) so that you could initialise the replicator before creating the models

Reply:
@Eunovo interesting, i her considered a callback function actually. I'll give it a try. 

Otherwise I was thinking of maybe having a getter and setter for the replicator, since the doDeltaSync method relies on it too.

Reply:
I think this makes no sense itself. Subscriptions are runtime setup only apparently. While delta makes sense to be forcefully called - we cannot really get live results from delta or subscriptions (all sadly needs to go thru the store - and notified to user back). 

I can do this quick fix, but this do not change that replication should be separate for subscriptions and deltas and play nice with store.